script/tool/lib/src/version_check_command.dart

```diff
@@ -268,16 +297,17 @@ ${indentation}HTTP response: ${pubVersionFinderResponse.httpResponse.body}
         allowedNextVersions[currentVersion] == NextVersionType.BREAKING_MAJOR) {
       printError('${indentation}Breaking change detected.\n'
           '${indentation}Breaking changes to platform interfaces are strongly discouraged.\n');
-      return false;
+      return _CurrentVersionState.invalidChange;
     }
-    return true;
+    return _CurrentVersionState.validChange;
   }
 
   /// Returns whether or not the pubspec version and CHANGELOG version for
   /// [plugin] match.
   Future<bool> _hasConsistentVersion(
     Directory package, {
     required Pubspec pubspec,
+    required bool versionChanged,
```

Comment:
Can `versionChanged` be calculated inside of this function?  The first 2 arguments make sense to me, this one isn't clear.  It seems like you wouldn't ask "has consistent version?" if you already know the "version hasn't changed".

We should at least update the docstring.  Also, based on the name it seems like a pure function but it is printing out stuff.

Reply:
I changed the name and docstring, so hopefully it's clearer what this is doing now (it's unfortunately hard to do pure functions in these checks a lot of the time, because verbose logging of everything that's happening is a feature in CI checks; failures where you don't get easily and clearly enough detail about why are a pain). I also changed the parameter name to be more specific about what version changed (the pubspec version) since that was, as evidenced by your question, ambiguous.

For an example of why we'd call this even when that hasn't changed: if a PR author forgets to actually update the version in `pubspec.yaml` but remembers to update `CHANGELOG.md` as if they had, we want that to fail CI. In that case the parameter here will be false, but the version consistency check will fail.